nncf/scopes.py

```diff
@@ -53,21 +56,15 @@ class IgnoredScope:
     ```
 
     **Note** Operation types must be specified according to the model framework.
-    """
 
-    def __init__(self,
-                 names: Optional[List[str]] = None,
-                 patterns: Optional[List[str]] = None,
-                 types: Optional[List[str]] = None):
-        """
-        :param names: List of ignored node names.
-        :param patterns: List of regular expressions that define patterns for
-            names of ignored nodes.
-        :param types: List of ignored operation types.
-        """
-        self.names = names if names is not None else []
-        self.patterns = patterns if patterns is not None else []
-        self.types = types if types is not None else []
+    :param names: List of ignored node names.
+    :param patterns: List of regular expressions that define patterns for names of
+        ignored nodes.
+    :param types: List of ignored operation types.
+    """
+    names: List[str] = field(default_factory=list)
```

Comment:
Only Python 3.10 supports this

https://docs.python.org/3/library/dataclasses.html#dataclasses.field

Reply:
Maybe something new in 3.10. But it is also supported in Python 3.7.

https://docs.python.org/3.7/library/dataclasses.html#dataclasses.field